src/core_ocean/shared/mpas_ocn_mesh.F

```diff
@@ -0,0 +1,842 @@
+         meshScaling,    &! general scaling of mesh size
+         meshDensity,    &! density of mesh
+         angleEdge        ! angle the edge normal makes with local east
+
+      ! Multiplicative masks and vectors for various conditions
+      real (kind=RKIND), dimension(:,:), allocatable :: &
+         edgeMask,       &! mask to denote active edges    with depth
+         cellMask,       &! mask to denote active cells    with depth
+         vertexMask,     &! mask to denote active vertices with depth
+         boundaryEdge,   &! mask for boundary edges    at each level
+         boundaryCell,   &! mask for boundary cells    at each level
+         boundaryVertex, &! mask for boundary vertices at each level
+         edgeSignOnCell, &! sign of edge contributions to a cell
+         edgeSignOnVertex,&! sign of edge contributions to a vertex
+         highOrderAdvectionMask ! mask for high order advection contributions
+
+      real (kind=RKIND), dimension(:,:), pointer :: &
+         weightsOnEdge,     &! weights on each edge
+         kiteAreasOnVertex, &! real (vertexDegree nVertices)
+         edgeTangentVectors,&! tangent unit vector at edge
+         edgeNormalVectors, &! normal  unit vector at edge
+         localVerticalUnitVectors, &! local unit vector iin vertical
+         advCoefs,          &! mesh-based advection coefficients
+         advCoefs3rd         ! mesh-based advection coeffs for high order
+
+      real (kind=RKIND), dimension(:,:,:), pointer :: &
+         derivTwo,          &! 2nd derivative of edge reconstruction polynomial
+         cellTangentPlane,  &! two vectors defining tangent plane at cell center
+         coeffs_reconstruct  ! coeffs for reconstructing vectors at cell centers
+
+   end type meshType
+
+   ! default mesh must be public for now to provide identical namespace
+   ! on CPU, accelerator
+   type (meshType), public :: &
+      ocnMesh              ! default mesh
+
+   !----------------------------------------------------------------------------
+   !
+   ! Public member functions
+   !
+   !----------------------------------------------------------------------------
+
+   public ::                     &
+      ocn_meshCreate,            &
+      ocn_meshUpdateFields,      &
+      ocn_meshDestroy
+
+!*******************************************************************************
+
+contains
+
+!*******************************************************************************
+!
+!  ocn_meshCreate
+!
+!> \brief Creates the primary ocean mesh data structure on both host and device
+!!  \author Rob Aulwes and Phil Jones
+!!  \date   14 Jan 2020
+!!  \details
+!!  This module creates and maintains a primary ocean mesh structure
+!!  and ensures all mesh variables are copied to an accelerator device
+!!  if needed.
+!
+!-------------------------------------------------------------------------------
+
+   subroutine ocn_meshCreate(domain)
+
+      ! Input arguments
+
+      type(domain_type) :: &
```

Comment:
I had intent there originally but the compiler complained, maybe because intent attributes for pointers is a F03 feature and may not be fully supported? Not sure, but had to remove it for a successful build.

Reply:
Ah, gotcha.